### ISSUE TITLE
Kalman Fusion for 3 signals

### ISSUE THREAD

**Issue by SutirthaChakraborty** [ZHYHenryZhang/pykffusion#1]
Hi,
I am a newbie, I want to fuse the 3 signal (x, y, Qom) together into one using Kalman Fusion. How can I do that. My signal file is attached.
[Dataset](https://drive.google.com/file/d/1CNsFBTflm-Wcpyu6929M2xe30_eA05Fr/view?usp=sharing)

Regards,
Suti

**Comment by henryzhangzhy** [comment]
Hi Suti,
I am not familiar with the context of music. Let's say the output is a signal y, do you mean you want to model y as a function of x, y, Qom and time t?

The fusion I am describing here is actually fusion information from multiple sensors. But your case sounds to me like from a single source but reduces the output features?

If that is the case, I would recommend you take a look at Kalman Filter (for linear case) or LSTM (using neural network), but again I am not familiar with the music case and not sure what your application is.

Best,
Henry

**Comment by SutirthaChakraborty** [comment]
Hi Henry,

I just want to fuse the x,y, and QoM into a single signal. They are generated from a video signal (motiongram) generated by using [Musical Gesture](https://github.com/fourMs/MGT-python)  and the code I implemented is here - [Colab Link](https://colab.research.google.com/drive/1-Rw_3X0sZJKsdiYUc-6b0TPf9WXS5m6a?usp=sharing)
I tried the LSTM model but didn't help much. 
Regards,
Suti

**Comment by henryzhangzhy** [comment]
> Hi Henry,
> 
> I just want to fuse the x,y, and QoM into a single signal. They are generated from a video signal (motiongram) generated by using [Musical Gesture](https://github.com/fourMs/MGT-python) and the code I implemented is here - [Colab Link](https://colab.research.google.com/drive/1-Rw_3X0sZJKsdiYUc-6b0TPf9WXS5m6a?usp=sharing)
> I tried the LSTM model but didn't help much.
> Regards,
> Suti

I see, but I am not sure what is good for your case. In the context of robotics applications, the fusion often are related by physical models. we fuse x from sensor A with x from sensor B to get a fused x. Or we fuse x from sensor A with y=h(x) in sensor B to get a fused x. In your case, fusing x, y, QoM are very different quantities that I don't know how are they meaningfully being fused. I don't have an idea about this kind of fusion.

**Comment by SutirthaChakraborty** [comment]
Yeah right. They are all motion data. I think they can be interpreted as a similar signal. That is why I am trying to do the 1st approach. How can I fuse the 3 individual arraylist (x,y, Qom)? 
It may work or it might fail. But still want to give it a try. 
Can you help me with the code?
